src/js/components/OptionalSettingsComponent.jsx

```diff
@@ -1,73 +1,79 @@
 var React = require("react/addons");
 
-var appValidator = require("../validators/appValidator");
-var FormGroupComponent = require("../components/FormGroupComponent");
+var AppFormErrorMessages = require("../validators/AppFormErrorMessages");
+var FormActions = require("../actions/FormActions");
+var StoreFormGroupComponent =
+  require("../components/StoreFormGroupComponent");
 
 var OptionalSettingsComponent = React.createClass({
   displayName: "OptionalSettingsComponent",
 
   propTypes: {
-    errors: React.PropTypes.array,
-    model: React.PropTypes.object.isRequired
+    errorIndices: React.PropTypes.object.isRequired,
+    fields: React.PropTypes.object.isRequired
   },
 
-  render: function () {
-    var model = this.props.model;
-    var errors = this.props.errors;
-    var helpMessage = "Comma-separated list of valid constraints. " +
-      "Valid constraint format is \"field:operator[:value]\".";
-    var attr = {};
-
-    if (model.constraints != null) {
-      attr.constraints = model.constraints.map(function (constraint) {
-        return constraint.join(":");
-      });
+  getErrorMessage: function (fieldId) {
```

Comment:
In #197 i decided to pass the `AppModalComponent.getErrorMessage()`  as a prop. I guess we could avoid the same duplication here?